tests/test_generators.py

```diff
@@ -176,3 +178,73 @@ def test_allowedbyaccesslevels_query_filter(mocker):
     )
 
     assert query_filter == []
+
+
+def test_disable_if_scalar_value(app):
+    generator = DisableIf(False)
+    assert generator.needs() == []
+    assert generator.excludes() == []
+    assert generator.query_filter() == []
+
+    generator = DisableIf(True)
+    assert generator.needs() == []
+    assert generator.excludes() == [any_user]
+    assert generator.query_filter().to_dict() == {"match_none": {}}
+
+
+def test_disable_if_function(app):
+    def _if_two_kwargs(**kwargs):
+        if len(kwargs) == 2:
+            return True
+        else:
+            return False
+
+    # The generator should only disable if 2 keyword arguments are supplied
+    generator = DisableIf(_if_two_kwargs)
+    assert generator.excludes(record=None) == []
+    assert generator.query_filter(record=None) == []
+
+    assert generator.excludes(record=None, second_arg=None) == [any_user]
+    assert generator.query_filter(record=None, second_arg=None).to_dict() == {
+        "match_none": {}
+    }
+
+    assert generator.excludes(record=None, second_arg=None, third_arg=None) == []
+    assert generator.query_filter(record=None, second_arg=None, third_arg=None) == []
+
+
+def test_disable_if_read_only(app):
+    generator = DisableIfReadOnly()
+
+    # Normal operation
+    app.config["RECORDS_PERMISSIONS_READ_ONLY"] = False
+    assert generator.excludes(record=None) == []
+    assert generator.query_filter(record=None) == []
+
+    # System is in read-only mode
+    app.config["RECORDS_PERMISSIONS_READ_ONLY"] = True
+    assert generator.excludes(record=None) == [any_user]
+    assert generator.query_filter(record=None).to_dict() == {"match_none": {}}
+
+
+def test_disable_if_read_only_function(app):
+    def _some_function(record, **kwargs):
+        if record is None and not kwargs.get("exception", False):
```

Comment:
minor: return direct the boolean value of `record is None and not kwargs.get("exception", False)` 

```
def _some_function(record, **kwargs):
    return record is None and not kwargs.get("exception", False)
```